pkg/kv/kvserver/client_split_test.go

```diff
@@ -1951,6 +1951,68 @@ func TestStoreSplitGCThreshold(t *testing.T) {
 	repl.AssertState(ctx, store.Engine())
 }
 
+func TestStoreSplitGCHint(t *testing.T) {
```

Comment:
Let's do a merge test too -- both when the LHS and RHS match, and when they don't (e.g. different value or missing value).